packages/extension-chakra-storefront/src/pages/product-detail/index.jsx

```diff
@@ -1,622 +1,86 @@
+            <Box
+                className="sf-product-detail-page"
+                layerStyle="page"
+                data-testid="product-details-page"
+            >
                 <Stack gap={16}>
-                    {!isProductASet && (
-                        <RecommendedProducts
-                            title={
-                                <FormattedMessage
-                                    defaultMessage="Complete the Set"
-                                    id="product_detail.recommended_products.title.complete_set"
-                                />
-                            }
-                            recommender={EINSTEIN_RECOMMENDERS.PDP_COMPLETE_SET}
-                            products={[product]}
-                            mx={{base: -4, md: -8, lg: 0}}
-                            shouldFetch={() => product?.id}
-                        />
-                    )}
-                    <RecommendedProducts
-                        title={
-                            <FormattedMessage
-                                defaultMessage="You might also like"
-                                id="product_detail.recommended_products.title.might_also_like"
-                            />
-                        }
-                        recommender={EINSTEIN_RECOMMENDERS.PDP_MIGHT_ALSO_LIKE}
-                        products={[product]}
-                        mx={{base: -4, md: -8, lg: 0}}
-                        shouldFetch={() => product?.id}
+                    <ProductDetails
+                        product={product}
+                        primaryCategory={primaryCategory}
+                        isProductASet={isProductASet}
+                        isProductABundle={isProductABundle}
+                        comboProduct={comboProduct}
+                        childProductRefs={childProductRefs}
+                        childProductSelection={childProductSelection}
+                        setChildProductSelection={setChildProductSelection}
+                        childProductOrderability={childProductOrderability}
+                        setChildProductOrderability={setChildProductOrderability}
+                        selectedBundleQuantity={selectedBundleQuantity}
+                        setSelectedBundleQuantity={setSelectedBundleQuantity}
+                        // Handlers
+                        handleAddToCart={handleAddToCart}
+                        handleAddToWishlist={handleAddToWishlist}
+                        handleProductSetAddToCart={handleProductSetAddToCart}
+                        handleProductBundleAddToCart={handleProductBundleAddToCart}
+                        handleChildProductValidation={handleChildProductValidation}
+                        // Loading states
+                        isProductLoading={isProductLoading}
+                        isBasketLoading={isBasketLoading}
+                        isWishlistLoading={isWishlistLoading}
```

Comment:
We may complain about	 the prop drilling here. But I kinda like it. 

It reminds me of the old days of `getProps`, which made it clear of where the data fetching was. And it wasn't spread out across many components.